src/pages/questions.js

```diff
@@ -20,18 +23,25 @@ export default function QuestionPage() {
 					content="Here you can see all the Questions"
 				/>
 			</Helmet>
+
 			<Button
 				greyButton={true}
 				onClick={() => {
 					setFilter(!filter);
 				}}
 			>
-				{!filter ? `Show Closed` : `Show Open`}
+				{filter ? `Show All Questions` : `Show My Questions`}
 			</Button>
 			<article>
 				{questions.map(question => {
-					if (question.answered === filter) {
-						return <QuestionComponent key={question.id} question={question} />;
+					if (filter ? question.userID === activeUser.userID : true) {
+						return (
+							<QuestionComponent
+								key={question.id}
+								ownedQuestion={question.userID === activeUser.userID ? true : false}
+								question={question}
+							/>
+						);
 					}
 					return null;
```

Comment:
Is there ever a case where `null` is returned?

Reply:
If the app works correctly no. But due to i use an "if" operator i need a "else" return value.
This could be prevented if i could use a ternary operator. Unfortunately i have no idea on how to get my filter to work properly without this if operator.

(Open for suggestions :D )

Reply:
Hi :) I do have a suggestion:
```js
import {useState} from 'react';
import {Helmet} from 'react-helmet';

import Button from '../components/Button';
import Layout from '../components/Layout';
import QuestionComponent from '../components/QuestionCard';
import {useStore} from '../hooks/useStore';

export default function QuestionPage() {
	const questions = useStore(state => state.questions);
	const activeUser = useStore(state => state.activeUser);
	const [filter, setFilter] = useState(false);

	questions.sort((a, b) => Number(a.answered) - Number(b.answered));

	const filteredQuestions = filter ? questions.filter(question => question.userID === activeUser.userID) : questions;

	return (
		<Layout>
			<Helmet>
				<title key="title">Question Area</title>
				<meta
					key="description"
					name="description"
					content="Here you can see all the Questions"
				/>
			</Helmet>

			<Button
				greyButton={true}
				onClick={() => {
					setFilter(!filter);
				}}
			>
				{filter ? `Show All Questions` : `Show My Questions`}
			</Button>
			<article>
				{filteredQuestions.map(question => {
						return (
							<QuestionComponent
								key={question.id}
								ownedQuestion={filter && (question.userID === activeUser.userID)}
								question={question}
							/>
						);
				})}
			</article>
		</Layout>
	);
}

```
Instead of doing the filter in an `if` statement, you could do it before the return statement:
```js
const filteredQuestions = filter ? questions.filter(question => question.userID === activeUser.userID) : questions;
```
This is a bit easier to read :)